core/src/main/scala/org/apache/spark/serializer/KryoSerializer.scala

```diff
@@ -109,6 +111,9 @@ class KryoSerializer(conf: SparkConf)
     kryo.register(classOf[SerializableJobConf], new KryoJavaSerializer())
     kryo.register(classOf[HttpBroadcast[_]], new KryoJavaSerializer())
     kryo.register(classOf[PythonBroadcast], new KryoJavaSerializer())
+    kryo.register(classOf[TaskMetrics], new KryoJavaSerializer())
+    kryo.register(classOf[DirectTaskResult[_]], new KryoJavaSerializer())
+    kryo.register(classOf[IndirectTaskResult[_]], new KryoJavaSerializer())
```

Comment:
I see, that does seem like a real problem. Let's move this discussion to the main thread so more people can see.